### PR TITLE
[Vulkan] Improved context factory

### PR DESCRIPTION
**Pull request by rokuz**

### CODE REVIEW

File: android/jni/com/mapswithme/vulkan/android_vulkan_context_factory.hpp [hunk 1]
```diff
@@ -48,6 +48,7 @@ class AndroidVulkanContextFactory : public dp::GraphicsContextFactory
   ANativeWindow * m_nativeWindow = nullptr;
   bool m_windowSurfaceValid = false;
   VkSurfaceKHR m_surface = 0;
+  VkFormat m_surfaceFormat;
```
Comment (darina) [inline android/jni/com/mapswithme/vulkan/android_vulkan_context_factory.hpp]: = VK_FORMAT_UNDEFINED?

File: android/jni/com/mapswithme/vulkan/android_vulkan_context_factory.cpp [hunk 2]
```diff
@@ -226,11 +224,45 @@ bool AndroidVulkanContextFactory::QuerySurfaceSize()
...
+    if (formats[chosenFormat].format == VK_FORMAT_R8G8B8A8_UNORM)
+      break;
+  }
+  if (chosenFormat == formatCount)
+  {
+    LOG_ERROR_VK("Any supported surface format wasn't found.");
+    return false;
+  }
+
+  if (!(surfaceCapabilities.supportedCompositeAlpha | VK_COMPOSITE_ALPHA_INHERIT_BIT_KHR))
```
Comment (darina) [inline android/jni/com/mapswithme/vulkan/android_vulkan_context_factory.cpp]: скорее &

File: android/jni/com/mapswithme/vulkan/android_vulkan_context_factory.hpp (hunk 1 above)
Comment (rokuz) [reply android/jni/com/mapswithme/vulkan/android_vulkan_context_factory.hpp]: done

File: android/jni/com/mapswithme/vulkan/android_vulkan_context_factory.cpp (hunk 2 above)
Comment (rokuz) [reply android/jni/com/mapswithme/vulkan/android_vulkan_context_factory.cpp]: done

**Pull request merged**